site/content/blog/making-the-most-out-of-clis.md

```diff
@@ -0,0 +1,93 @@
+
+### Managing your `kubeconfig` and contexts
+The set of commands under `kubectl config` allow us to manage and modify the configurations we use while connecting to our clusters. These configurations include values which indicate which cluster the tool interacts with, how we authenticate with the cluster and default namespace used.
+This is used as the default configuration for connecting to a cluster by other tools on the host - more often than not. And `kubectl` allows us to seamlessly modify and switch between multiple "contexts" like these.
+
+### Interacting with nodes
+`kubectl` has a set of powerful commands that let us cordon and uncordon nodes (making them available or not available for scheduling). These commands can also help users drain resources from a node so that it can be cordoned safely. Nodes can be "tainted" with certain key-values, which allow only pods that tolerate those "taints" to be scheduled on them.
+Learn more about these commands [here](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#-strong-cluster-management-strong-).
+
+### Dive into your containers and applications
+`kubectl` allows us to attach to processes containers running on pods and execute commands inside them easily. Users can observe the resource consumption on a pod as well.
+// might remove
+Proxying and port-forwarding options let a pod or the K8s API server serve HTTP content to localhost. Letting users debug their applications easily
+
+## `kapp`
+`kapp` is a CLI team that helps users manage their workloads effectively on the cluster. This involves defining interactions between resources declaratively, deploying resources reliably and updating them safely and confidently.
+
+But wait isn't this something `kubectl apply` or `kubectl patch` will help us do for resource manifests?
+
+Well yes, but `kapp` goes a step further, it helps users order resources in their manifests, apply changes safely and predictably, and then goes on to wait for the resources to reach their desired state. (You could take a deep dive into this over [here](https://carvel.dev/kapp/docs/latest/))
+
+Let's see how `kapp` makes our lives easier.
+
+### Grouping related resources
+`kapp` lets users group and [order multiple resources](https://carvel.dev/kapp/docs/latest/apply-ordering/#example) easily. This lets us do tasks like "create a Deployment after a Job has finished" declaratively in our manifests.
+`kapp` also caters to other use cases, like [versioning resources](https://carvel.dev/kapp/docs/latest/diff/#versioned-resources) and defining [create/update/delete strategies](https://carvel.dev/kapp/docs/latest/apply/#controlling-apply-via-resource-annotations) for resources in the abstract "application". These features allow users to efficiently, declaratively and effectively manage resources to be deployed on clusters.
```

Comment:
nit: efficiently and effectively seem a lot similar while skimming through the blog and could be a bit confusing.
```suggestion
`kapp` also caters to other use cases, like [versioning resources](https://carvel.dev/kapp/docs/latest/diff/#versioned-resources) and defining [create/update/delete strategies](https://carvel.dev/kapp/docs/latest/apply/#controlling-apply-via-resource-annotations) for resources in the abstract "application". These features allow users to efficiently, declaratively and adequately manage resources to be deployed on clusters.
```
OR
```suggestion
`kapp` also caters to other use cases, like [versioning resources](https://carvel.dev/kapp/docs/latest/diff/#versioned-resources) and defining [create/update/delete strategies](https://carvel.dev/kapp/docs/latest/apply/#controlling-apply-via-resource-annotations) for resources in the abstract "application". These features allow users to efficiently and declaratively manage resources to be deployed on clusters.
```
should also be fine.

Reply:
I am gonna go with declaratively, sounds good!